frontend/app/models/user.rb

```diff
@@ -59,4 +62,42 @@ def self.become_user(context, username)
     end
   end
 
+
+  def self.recover_password(email)
+    uri = JSONModel(:user).uri_for("recover-password")
+
+    begin
+      response = JSONModel::HTTP.post_form(uri, email: email)
+      message = ASUtils.json_parse(response.body)
+      if response.code == '200'
+        return {status: :success}
+      else
+        msg = case message['error']
+              when "UserMailer::MailError"
+                I18n.t("user._frontend.messages.password_recovery_fail")
+              when "NotFoundException"
+                I18n.t("user._frontend.messages.error_not_found", email: email)
```

Comment:
@quoideneuf Do we want to report that an entered e-mail address does not correspond to an account? Usually this would be silently ignored to not enable account checking by trying out e-mail addresses.

Reply:
Though it does potentially make things a little more confusing for users, I'd tend to agree with you on this. I'll chat with the team  here and see what they think.